File: count-symmetric-integers.ts
{
    function countSymmetricIntegers(low: number, high: number): number {
        let c = 0;

        for(;low <= high; low++) {
            const char = low.toString();
            const first = char.slice(0, char.length / 2).split("").reduce((a,b) => a + +b , 0)
            const second = char.slice(char.length / 2, char.length).split("").reduce((a,b) => a + +b , 0)
            
            if(char.length % 2 === 0 && first === second) {
                c++
            }
        }

        return c
    };
    
    console.log(countSymmetricIntegers(1,100))
}

File: count-symmetric-integers.rs
pub fn count_symmetric_integers(low: i32, high: i32) -> i32 {
    let mut count = 0;

    for num in low..=high {
        let s = num.to_string();
        let len = s.len();

        if len % 2 == 0 {
            let half = len / 2;
            let first_half = s[..half]
                .chars()
                .map(|c| c.to_digit(10).unwrap() as i32)
                .sum();
            let second_half = s[half..]
                .chars()
                .map(|c| c.to_digit(10).unwrap() as i32)
                .sum();

            if first_half == second_half {
                count += 1;
            }
        }
    }

    count
}
